sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -856,7 +857,90 @@ trait GpuAggregateWindowFunction[T <: Aggregation with RollingAggregation[T]]
   def windowAggregation(inputs: Seq[(ColumnVector, Int)]): AggregationOnColumn[T]
 }
 
-case class GpuRowNumber() extends GpuAggregateWindowFunction[RowNumberAggregation] {
+trait BatchedRunningWindowFixer extends AutoCloseable {
+  /**
+   * Save any state needed from the previous output that will be needed to fix up the next batch.
+   * You don't need to worry about the partition columns. Those will be saved separately.
+   * @param finalOutputColumn the output of fixup.
+   */
+  def updateState(finalOutputColumn: GpuColumnVector): Unit
+
+  /**
+   * Fix up `windowedColumnOutput` with any stored state from previous batches.
+   * @param samePartitionMask a boolean mask with no nulls in it. `true` indicates the row
+   *                          is for the same partition by keys that was the last row in the
```

Comment:
ok so technically you can an inbound `windowedColumnOutput` with keys `[key1, key1, key2, key2]`, where `samePartitionMask` would be `[false, true, false, true]`. From the header, I am not entirely sure if `previous batches` means previous calls to `fixUp`.

Reply:
would be nice to note when `sameParttionMask` can be a vector, and when it is a scalar.